### PR TITLE
Fixes #32584 - drop flot charts

### PR DESCRIPTION
**Pull request by ezr-ondrej**
All uses of flot charts have been deprecated and are rendering React
counterparts. We can remove flot charts now :boom:

~~Need to wait for the last deprecation #8518~~

### CODE REVIEW

File: app/assets/javascripts/charts.js [hunk 1]
```diff
@@ -1,264 +1 @@
...
-  );
-});
-
-$(window).on('resize', refreshCharts);
-
-function refreshCharts() {
-  $('.statistics-bar:visible').flot_bar();
-  $('.statistics-chart:visible').flot_chart();
-}
+tfm.tools.deprecate('charts.js file', 'React chart components', '3.1');
```
Comment (tbrisker) [inline app/assets/javascripts/charts.js]: Should we just remove this file completely? anyone attempting to use it would fail anyways

Comment (ezr-ondrej) [reply app/assets/javascripts/charts.js]: Yeah, but they will get this deprecation message to see why it failed, but I guess they can find the info other ways :+1:

File: app/assets/stylesheets/charts.scss [hunk 2]
```diff
@@ -1,157 +0,0 @@
...
-  padding: 15px;
-  min-width: 280px;
-  min-height: 100px;
-  margin-bottom: 15px;
-
-  h2,
-  h4 {
-    font-weight: normal;
-  }
-}
```
Comment (tbrisker) [inline app/assets/stylesheets/charts.scss:13]: This class is still used in some places

File: app/assets/stylesheets/charts.scss [hunk 3]
```diff
@@ -1,157 +0,0 @@
...
-
-#host-show {
-  .statistics-chart {
-    height: 230px;
-  }
-
-  .chart {
-    height: 250px;
-  }
-}
```
Comment (tbrisker) [inline app/assets/stylesheets/charts.scss:128]: I thinks theses classes as well

**Review by tbrisker — CHANGES_REQUESTED** [review]
some of the css classes are still used in other places, please move them to one of the other stylesheets

File: app/assets/stylesheets/charts.scss (hunk 2 above)
Comment (ezr-ondrej) [reply app/assets/stylesheets/charts.scss:13]: I hope I've done all that are still needed.

File: app/assets/stylesheets/charts.scss (hunk 3 above)
Comment (ezr-ondrej) [reply app/assets/stylesheets/charts.scss:128]: These just made the current charts look bad.
I've added commit before the drop that migrates to rendering the React components for Host detail page charts directly and makes them look better at least spacialy.

**Comment by ezr-ondrej** [comment]
It now also include direct rendering of Host Detail page charts as separate commit, so we do not have deprecation warnings rendered by core.
I hope I've taken care of all the still usefull styles in charts, but styles are hard, so not sure, but I'd prefer droping and fixing some minor visual issues if those appear.

File: app/views/hosts/_resources.html.erb [hunk 4]
```diff
@@ -1 +1,5 @@
-<%= flot_chart('resource_graph', '', _("Number of Events"), resources_chart(params["range"].to_i.days.ago)) %>
+<%= react_component('AreaChart', id: 'resource_graph',
+                                 xAxisLabel: '',
+                                 yAxisLabel: _('Number of Events'),
+                                 data: resources_chart_data(params["range"].to_i.days.ago),
+                                 size: { height: 270 }) %>
```
Comment (tbrisker) [inline app/views/hosts/_resources.html.erb]: This (and the other one) used to be 250 previously, for me at least 270 causes the page to overflow.

**Review by tbrisker — COMMENTED** [review]
Looks good @ezr-ondrej , great cleanup! one inline comment, other than that it should be good to go. can you please post on discourse notifying plugin maintainers that they should update their plugins and maybe even list the removed functions?

**Comment by ezr-ondrej** [comment]
I've created a discourse anouncement: https://community.theforeman.org/t/warning-flot-charts-are-going-away/24318

**Review by tbrisker — APPROVED** [review]
Thanks @ezr-ondrej !

**Pull request merged**